mazegame/Assets/Scripts/Maze/MazeGenerator.cs

```diff
@@ -0,0 +1,176 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class MazeGenerator : MonoBehaviour
+{
+
+    public int mazeWidth;
+    public int mazeHeight;
+    public string mazeSeed;
+
+    public Sprite floorSprite;
+    public Sprite roofSprite;
+    public Sprite wallSprite;
+    public Sprite wallCornerSprite;
+
+    public MazeSprite mazeSpritePrefab;
+
+    System.Random mazeRG;
+
+    Maze maze;
+
+    public Vector3 mazeGoalPosition;
+
+    public delegate void MazeReadyAction();
+    public static event MazeReadyAction OnMazeReady;
+
+
+    void Start()
+    {
+        mazeRG = new System.Random(mazeSeed.GetHashCode());
+
+        if (mazeWidth % 2 == 0)
+            mazeWidth++;
+
+        if (mazeHeight % 2 == 0)
+        {
+            mazeHeight++;
+        }
+
+        maze = new Maze(mazeWidth, mazeHeight, mazeRG);
+        maze.Generate();
+
+        mazeGoalPosition = maze.GetGoalPosition();
+
+        DrawMaze();
+
+        if (OnMazeReady != null)
+        {
+            OnMazeReady();
+        }
+    }
+
+    void DrawMaze()
+    {
+        for (int x = 0; x < mazeWidth; x++)
+        {
+            for (int y = 0; y < mazeHeight; y++)
+            {
+                Vector3 position = new Vector3(x, y);
+
+                if (maze.Grid[x, y] == true)
```

Comment:
a comment here would be good. Not 100% sure what is being checked. Is it the edge of the maze? A corner?

Reply:
// Check if the cell is a path, if it isn't, add it.